client/packages/cli/src/commands/estimate.ts

```diff
@@ -0,0 +1,120 @@
+import ora from "ora"
+import { ethGasFeeEstimationUtils } from "@t3rn/sdk"
+import { ExtrinsicSchema, SpeedMode, SpeedModes } from "@/schemas/extrinsic.ts"
+import { getConfig } from "@/utils/config.ts"
+import { validate } from "@/utils/fns.ts"
+import { colorLogMsg, log } from "@/utils/log.ts"
+import { readSfxFile } from "@/utils/sfx.ts"
+import { Args } from "@/types.ts"
+import {
+  SideEffect,
+  SideEffectAction,
+  SideEffectActions,
+} from "@/schemas/sfx.ts"
+
+export const spinner = ora()
+
+export const handleEstimateFees = async (args: Args<"sfx" | "export">) => {
+  const config = getConfig()
+  if (!config) {
+    process.exit(1)
+  }
+
+  if (!args.sfx) {
+    log("ERROR", "No sfx file provided, use --sfx <file-path>")
+    process.exit(1)
+  }
+
+  const unvalidatedExtrinsic = readSfxFile(args.sfx)
+
+  if (!unvalidatedExtrinsic) {
+    process.exit(1)
+  }
+
+  const extrinsic = validate(ExtrinsicSchema, unvalidatedExtrinsic, {
+    configFileName: args.sfx,
+  })
+
+  if (!extrinsic) {
+    process.exit(1)
+  }
+
+  spinner.text = "Estimating fees..."
+  spinner.start()
+
+  await Promise.allSettled(
+    extrinsic.sideEffects.map((sfx, i) =>
+      estimateFees(`#${i}`, sfx, extrinsic.speed_mode)
+    )
+  )
+
+  spinner.stop()
+}
+
+const estimateFees = async (
+  tag: string,
+  sideEffect: SideEffect,
+  speedMode: SpeedMode
+) => {
+  const withArgs = (fn: (...args: unknown[]) => unknown) =>
+    fn(tag, sideEffect, speedMode)
+
+  switch (sideEffect.target) {
+    case "roco":
```

Comment:
"polk", "ksma"

Reply:
Fixed, refer to commit [853a45d](https://github.com/t3rn/t3rn/pull/1091/commits/853a45d48f5b9af932bc6b7bd2fbed93c87e34fa).